### PR TITLE
gh-106584: Fix exit code for unittest in Python 3.12

### PR DESCRIPTION
**Pull request by EliseevEgor**
Increase the number of running tests only if a test hasn't been skipped.

* Issue: gh-106584

### CODE REVIEW

**Review by sobolevn — APPROVED** [review]
LGTM, but please add a NEWS entry!

File: Misc/NEWS.d/next/Library/2023-07-11-08-56-40.gh-issue-106584.g-SBtC.rst [hunk 1]
```diff
@@ -0,0 +1,2 @@
+Fix exit code for `unittest` in Python 3.12 or greater if all
```
Comment (sunmy2019) [inline Misc/NEWS.d/next/Library/2023-07-11-08-56-40.gh-issue-106584.g-SBtC.rst]:
```suggestion
Fix exit code for `unittest` in Python 3.12 or greater if all
```

File: Lib/unittest/case.py [hunk 2]
```diff
@@ -617,6 +616,9 @@ def run(self, result=None):
                 _addSkip(result, self, skip_why)
                 return result
 
+            # Increase the number of running tests only if it hasn't been skipped
```
Comment (sunmy2019) [inline Lib/unittest/case.py]: it's not "running tests", it's "tests run". Also, line length > 80


```suggestion
            # Increase the number of tests only if it hasn't been skipped
```

File: Misc/NEWS.d/next/Library/2023-07-11-08-56-40.gh-issue-106584.g-SBtC.rst [hunk 3]
```diff
@@ -0,0 +1,2 @@
...
```
Comment (sobolevn) [inline Misc/NEWS.d/next/Library/2023-07-11-08-56-40.gh-issue-106584.g-SBtC.rst]:
```suggestion
Fix exit code for ``unittest`` if all

```

**Review by sobolevn — COMMENTED** [review]
No need to specify the version info in the NEWS.

**Comment by EliseevEgor** [comment]
Could you help me with `Azure Pipelines PR` test?
All tests are passed on my machine. Maybe you know what I should do to make it pass? Or why it fails?

**Comment by sunmy2019** [comment]
We got some failed tests. Most are unrelated, but we might need to rule them out.

**Comment by exarkun** [comment]
It doesn't seem like this ought to have been backported.  It includes an incompatible change in the long-established protocol between a test runner and a test results object.  One consequence is that it has broken trial's multiprocess test runner (https://github.com/twisted/twisted/issues/12052).

**Comment by yilei** [comment]
FYI-- This also broke `absl.testing` where it tracks test runtime with `startTest` and `stopTest`: https://github.com/abseil/abseil-py/blob/2af184bfcdaa6156a5a098923a9561bc4b7d988c/absl/testing/xml_reporter.py#L352-L365

We should be able to workaround it, but just wanted to point this out. It seems strange for `stopTest` to be called without a `startTest`.

**Pull request merged**